Fix Flask deprecation warning

jedcunningham opened the pull request:
`app.json_encoder` has been deprecated in favor of
`app.json_provider_class`, and the former is being removed in Flask 2.3.

jedcunningham committed:
Fix Flask deprecation warning

`app.json_encoder` has been deprecated in favor of
`app.json_provider_class`, and the former is being removed in Flask 2.3.

ashb reviewed (changes requested):
If I'm looking at Flask right this only works for 2.2+, but we have a dep on 2.0...2.3

https://github.com/pallets/flask/pull/4692

jedcunningham committed:
Bump min Flask version to 2.2

uranusjr committed:
Don't pass datetime to JSON serializer

Also convert those parametrized calls to pytest. This helped me find the
test I was looking for.
